cli/tests/xcresult.rs

```diff
@@ -0,0 +1,64 @@
+use trunk_analytics_cli::xcresult::XCResultFile;
+
+#[cfg(target_os = "macos")]
+#[test]
+fn test_xcresult_with_valid_path() {
+    let path_str = "tests/data/test1.xcresult";
+    let xcresult = XCResultFile::new(path_str.to_string());
+    assert!(xcresult.is_ok());
+
+    let junit = xcresult.unwrap().junit();
+    let expected_path = "tests/data/test1.junit";
+    // open the expected file
+    let expected = std::fs::read_to_string(expected_path).unwrap();
+    assert_eq!(String::from_utf8(junit).unwrap(), expected);
+}
+
+#[cfg(target_os = "macos")]
+#[test]
+fn test_xcresult_with_invalid_path() {
+    let path_str = "tests/data/test2.xcresult";
+    let xcresult = XCResultFile::new(path_str.to_string());
+    assert!(xcresult.is_err());
+    assert_eq!(
+        xcresult.err().unwrap().to_string(),
+        "failed to get absolute path"
+    );
+}
+
+#[cfg(target_os = "macos")]
+#[test]
+fn test_xcresult_with_invalid_xcresult() {
+    let path_str = "tests/data/test3.xcresult";
+    let xcresult = XCResultFile::new(path_str.to_string());
+    assert!(xcresult.is_err());
+    assert_eq!(
+        xcresult.err().unwrap().to_string(),
+        "failed to parse json from xcrun output"
+    );
+}
+
+#[cfg(target_os = "macos")]
+#[test]
+fn test_complex_xcresult_with_valid_path() {
+    let path_str = "tests/data/test4.xcresult";
+    let xcresult = XCResultFile::new(path_str.to_string());
+    assert!(xcresult.is_ok());
+
+    let junit = xcresult.unwrap().junit();
+    let expected_path = "tests/data/test4.junit";
+    // open the expected file
+    let expected = std::fs::read_to_string(expected_path).unwrap();
+    assert_eq!(String::from_utf8(junit).unwrap(), expected);
+}
+
+#[cfg(target_os = "linux")]
+#[test]
+fn test_xcresult_with_valid_path_invalid_os() {
+    let path_str = "tests/data/test1.xcresult";
+    let xcresult = XCResultFile::new(path_str.to_string());
+    assert_eq!(
+        xcresult.err().unwrap().to_string(),
+        "xcrun is only available on macOS"
+    );
+}
```

Comment:
The actual tests. Commenting since the test data files are obfuscating this at the bottom.